frontend/js/src/artist-similarity/src/artist-search/SearchBox.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useEffect, useMemo } from "react";
+import SearchDropdown from "./SearchDropdown";
+import ArtistLookup from "./ArtistLookup";
+import { ArtistType } from "./ArtistLookup";
+import { faMagnifyingGlass, faMinus, faPlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import "./SearchBox.css";
+interface SearchBoxProps {
+    currentsimilarArtistsLimit: number;
+    onsimilarArtistsLimitChange: (similarArtistsLimit: number) => void;
+    currentArtistMbid: string;
+    onArtistChange: (artist_mbid: string) => void;
+}
+
+const SearchBox = (props: SearchBoxProps) => {
+    // State to store the search results (list of artists)
+    const [searchResults, setSearchResults] = React.useState<Array<ArtistType>>([]);
+    const [searchQuery, setSearchQuery] = React.useState<string>("");
+    const [similarArtistsLimit, setsimilarArtistsLimit] = React.useState<number>(props.currentsimilarArtistsLimit);
```

Comment:
The naming here is quite critical.
But as suggested in a comment below, you most likely don't want to both pass the limit as props and also set it in this component's state at the same time.
```suggestion
    const [similarArtistsLimit, setSimilarArtistsLimit] = React.useState<number>(props.currentsimilarArtistsLimit);
```

Reply:
Fixed.